evm/utils.js

```diff
@@ -642,153 +636,200 @@ function wasEventEmitted(receipt, contract, eventName) {
-            const chain = config.chains[chainName.toLowerCase()];
-
-            if (chainsToSkip.includes(chain.name.toLowerCase()) || chain.status === 'deactive') {
-                printWarn('Skipping chain', chain.name);
-                return Promise.resolve();
+    let failedChains = {};
+    let promisedChainsResults = [];
+    let results = [];
+    for (const chain of chains) {
+        const chainTask = asyncChainTask(processCommand, axelarConfig, chain, chainsSnapshot, options);
+        if (options.parallel) {
+            promisedChainsResults.push(chainTask);
+        } else {
+            const { result, loggerError, chainName } = await chainTask;
+            if (result !== undefined) {
+                results.push(result);
             }
-
-            return new Promise((resolve) => {
-                cmds[chainCommandIndex + 1] = chainName;
-
-                exec(cmds.join(' '), { stdio: 'inherit' }, (error, stdout) => {
-                    printInfo('-------------------------------------------------------');
-                    printInfo(`Logs for ${chainName}`, stdout);
-
-                    if (error) {
-                        printError(`Error while running script for ${chainName}`, error);
-                    } else {
-                        successfullChains.push(chainName);
-                        printInfo(`Finished running script for chain`, chainName);
-                    }
-
-                    resolve();
-                });
-            });
-        };
-
-        await Promise.all(chains.map(executeChain));
-
-        if (save) {
-            for (const chainName of successfullChains) {
-                config.chains[chainName.toLowerCase()] = loadParallelExecutionConfig(options.env, chainName);
+            if (loggerError) {
+                failedChains[chainName] = loggerError;
             }
-
-            saveConfig(config, options.env);
         }
+    }
 
-        return;
+    if (options.parallel) {
+        const resultsWithErrLogs = await Promise.allSettled(promisedChainsResults);
+        results = resultsWithErrLogs
+            .filter((promiseResult) => promiseResult.status === 'fulfilled')
+            .map((promiseResult) => promiseResult.value.result)
+            .filter((chainResult) => chainResult !== undefined);
+        failedChains = resultsWithErrLogs.reduce((acc, promiseResult) => {
+            if (promiseResult.status === 'rejected') {
+                acc[promiseResult.reason.chainName || 'unknown'] = promiseResult.reason.message;
+            } else if (promiseResult.value.loggerError) {
+                acc[promiseResult.value.chainName] = promiseResult.value.loggerError;
+            }
+            return acc;
+        }, {});
     }
 
-    const chainsSnapshot = JSON.parse(JSON.stringify(config.chains));
+    printInfo(
+        'Succeeded chains',
+        chains
+            .filter((chain) => !Object.keys(failedChains).includes(chain.name))
```

Comment:
```suggestion
            .filter((chain) => !failedChains[chain.axelarId])
```
why do a linear search instead of a lookup?